.github/actions/cache-source-checkout/action.yml

```diff
@@ -0,0 +1,57 @@
+name: "Cache Source Checkout"
+description: "GitHub Action to cache checkout of source repos"
```

Comment:
Question: why is this necessary for CI? Why not check out each time?

Reply:
Some repo projects can take quite a while to clone everything, even if it's only a shallow clone. So it can be faster to just cache the one-off checkout that triggered the workflow, so that each job in the workflow (that runs on a new/different runner VM) can skip the time in re-checking out the same code. Also, because you want to use a repos yaml file that may pin the source code to a tag or branch, rather than a fixed commit sha, it is possible for checkouts to differ of time and introduce race conditions, resulting in non-consistent sources of the duration of a given workflow.